src/main/java/org/junitpioneer/jupiter/json/JacksonJsonConverter.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2016-2021 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * http://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junitpioneer.jupiter.json;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.UncheckedIOException;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+/**
+ * A {@link JsonConverter} using Jackson 2 {@link ObjectMapper} to perform the conversion
+ */
+class JacksonJsonConverter implements JsonConverter {
```

Comment:
Can you explain why we need this class and `JacksonNode`?

Reply:
I also don't know why we need `JsonConverter` and `JsonConverterProvider`. 🙈 Halp?

Reply:
The idea of the `JsonConverter` and `Node` is to provide an extra layer of abstraction over the Json Parsers. This would allow for using different parser implementations, such as [GSON](https://github.com/google/gson), [Johnzon](https://johnzon.apache.org/), [Moshi](https://github.com/square/moshi), without any changes in the main logic. 

This was inspired by the [`NodeFactory`](https://github.com/lukas-krecan/JsonUnit/blob/master/json-unit-core/src/main/java/net/javacrumbs/jsonunit/core/internal/NodeFactory.java) from the JsonUnit project.

See also `JsonConverterProvider` in this PR.

I will add some Javadoc to `JsonConverter` and `Node` to make it clearer.

Reply:
Got it!